backend/contracts/OasisReward.sol

```diff
@@ -1,19 +1,31 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import {ERC721, ERC721Enumerable} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
-import {Base64} from "@openzeppelin/contracts/utils/Base64.sol";
-
+import {
+    ERC721, ERC721Enumerable
+} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import {
+    Base64
+} from "@openzeppelin/contracts/utils/Base64.sol";
+
+/**
+ * @title OasisReward
+ * @notice An ERC721 contract that allows minting of NFT rewards for 
+    Oasis Quizzes.
+ * Contains functions to mint new NFTs, generate SVG images, 
+    and retrieve token data.
+ */
 contract OasisReward is ERC721Enumerable {
-    // Mapping from token ID to IPFS URI
-    mapping(uint256 => string) private _tokenURIs;
+    // Mapping from token ID to json metadata index
+    mapping(uint256 => uint256) private _hashIDs;
```

Comment:
Token ID is just an auto-increment number right? Why not simply use a list then? Also, use natspec to document contract members.

```suggestion
    /// Mapping from the token ID to its metadata JSON hash
    bytes32[] private _tokenURIHashes;
```

Reply:
Changed to byte array.